test/trento/application/projectors/database_projector_test.exs

```diff
@@ -264,4 +265,41 @@ defmodule Trento.DatabaseProjectorTest do
 
     assert deregistered_at == projection.deregistered_at
   end
+
+  test "should remove a database instance from the read model after a deregistration" do
+    deregistered_at = DateTime.utc_now()
+
+    insert(:database, id: sap_system_id = Faker.UUID.v4())
+    insert_list(4, :database_instance)
+
+    %{instance_number: instance_number, host_id: host_id} =
+      insert(:database_instance, sap_system_id: sap_system_id)
+
+    event = %DatabaseInstanceDeregistered{
+      instance_number: instance_number,
+      host_id: host_id,
+      sap_system_id: sap_system_id,
+      deregistered_at: deregistered_at
+    }
+
+    ProjectorTestHelper.project(DatabaseProjector, event, "database_projector")
+
+    assert nil ==
+             Repo.get_by(DatabaseInstanceReadModel,
+               sap_system_id: sap_system_id,
+               instance_number: instance_number,
+               host_id: host_id
+             )
+
+    query = from(d in DatabaseInstanceReadModel, select: fragment("count(?)", d.sap_system_id))
```

Comment:
Simpler this way maybe?
Why not simply 
```
assert 4 ==
    DatabaseInstanceReadModel
    |> Replo.all()
    |> Enum.count()
```

PD: Please, be strict with the order of the asserts: `assert expected == real`